operate/client/src/App/Dashboard/v2/InstancesByProcess/index.tsx

```diff
@@ -76,110 +82,149 @@ const InstancesByProcess: React.FC = observer(() => {
-              <Details
-                processName={processName || bpmnProcessId}
-                processes={processes}
-              />
+            id: rowKey,
+            instance: (
+              <LinkWrapper
+                to={Locations.processes({
+                  process: processDefinitionId,
+                  version: version.toString(),
+                  active: true,
+                  incidents: true,
+                  ...(totalInstancesCount === 0
+                    ? {
+                        completed: true,
+                        canceled: true,
+                      }
+                    : {}),
+                  ...(isMultiTenancyEnabled
+                    ? {
+                        tenant: normalizedTenantId,
+                      }
+                    : {}),
+                })}
+                onClick={() => {
+                  panelStatesStore.expandFiltersPanel();
+                  tracking.track({
+                    eventName: 'navigation',
+                    link: 'dashboard-process-instances-by-name-all-versions',
+                  });
+                }}
+                title={getAccordionTitle({
+                  processName: name,
+                  instancesCount: totalInstancesCount,
+                  versionsCount,
+                  ...(isMultiTenancyEnabled
+                    ? {
+                        tenant: tenantName,
+                      }
+                    : {}),
+                })}
+              >
+                <InstancesBar
+                  label={{
+                    type: 'process',
+                    size: 'medium',
+                    text: getAccordionLabel({
+                      name,
+                      instancesCount: totalInstancesCount,
+                      versionsCount,
+                      ...(isMultiTenancyEnabled
+                        ? {
+                            tenant: tenantName,
+                          }
+                        : {}),
+                    }),
+                  }}
+                  incidentsCount={activeInstancesWithIncidentCount}
+                  activeInstancesCount={activeInstancesWithoutIncidentCount}
+                  size="medium"
+                />
+              </LinkWrapper>
             ),
           };
         },
-        {},
       )}
+      expandedContents={(result.data?.items ?? []).reduce<{
+        [key: string]: React.ReactElement<{tabIndex: number}>;
+      }>((accumulator, item: ProcessDefinitionInstanceStatistics) => {
+        const {
+          processDefinitionId,
+          latestProcessDefinitionName,
+          hasMultipleVersions,
+          tenantId,
+        } = item;
+
+        const normalizedTenantId = tenantId ?? '<default>';
+        const rowKey = generateProcessKey(
+          processDefinitionId,
+          normalizedTenantId,
+        );
+
+        if (!hasMultipleVersions) {
+          return accumulator;
+        }
+
+        return {
```

Comment:
🔧 The performance profile of the reduce function could be improved quite a bit, by mutating the `accumulator` object instead of spreading into a new object each iteration.